### PR TITLE
JTC touchups #abi-breaking

### PR DESCRIPTION
**Pull request by bmagyar**
* Renamed the actionserver callbacks to better describe the reason
* Moved the topic callback to a member function for testability and ease of finding it
* Comments cleanup

### CODE REVIEW

File: joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.hpp [hunk 1]
```diff
@@ -213,8 +217,7 @@ class JointTrajectoryController : public controller_interface::ControllerInterfa
   rclcpp_action::CancelResponse cancel_callback(
```
Comment (destogl) [inline joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.hpp:217]:
```suggestion
  rclcpp_action::CancelResponse cancel_goal_callback(
```

File: joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.hpp [hunk 2]
```diff
@@ -205,6 +205,10 @@ class JointTrajectoryController : public controller_interface::ControllerInterfa
   rclcpp::TimerBase::SharedPtr goal_handle_timer_;
   rclcpp::Duration action_monitor_period_ = rclcpp::Duration(50ms);
 
+  // callback for topic interface
+  JOINT_TRAJECTORY_CONTROLLER_PUBLIC
+  void topic_callback(const std::shared_ptr<trajectory_msgs::msg::JointTrajectory> msg);
+
   // callbacks for action_server_
   JOINT_TRAJECTORY_CONTROLLER_PUBLIC
   rclcpp_action::GoalResponse goal_callback(
```
Comment (destogl) [inline joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.hpp:214]:
```suggestion
  rclcpp_action::GoalResponse cancel_goal_callback(
```

File: joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.hpp [hunk 3]
```diff
@@ -213,8 +217,7 @@ class JointTrajectoryController : public controller_interface::ControllerInterfa
   rclcpp_action::CancelResponse cancel_callback(
     const std::shared_ptr<rclcpp_action::ServerGoalHandle<FollowJTrajAction>> goal_handle);
   JOINT_TRAJECTORY_CONTROLLER_PUBLIC
-  void feedback_setup_callback(
-    std::shared_ptr<rclcpp_action::ServerGoalHandle<FollowJTrajAction>> goal_handle);
+  void accepted_callback(std::shared_ptr<rclcpp_action::ServerGoalHandle<FollowJTrajAction>> goal_handle);
```
Comment (destogl) [inline joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.hpp:220]:
```suggestion
  void accept_goal_callback(std::shared_ptr<rclcpp_action::ServerGoalHandle<FollowJTrajAction>> goal_handle);
```

**Review by jaron-l — APPROVED** [review]
Looks like good clean up!

**Review by destogl — APPROVED** [review]
Thanks and proposal for even more explicit names!

File: joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.hpp (hunk 1 above)
Comment (destogl) [inline joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.hpp:217]:
```suggestion
  rclcpp_action::CancelResponse cancel_goal_callback(
```

File: joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.hpp (hunk 2 above)
Comment (bmagyar) [reply joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.hpp:214]:
```suggestion
  rclcpp_action::GoalResponse receive_goal_callback(
```

File: joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.hpp (hunk 1 above)
Comment (bmagyar) [reply joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.hpp:217]:
```suggestion
  rclcpp_action::CancelResponse goal_cancelled_callback(
```

File: joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.hpp (hunk 3 above)
Comment (bmagyar) [reply joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.hpp:220]:
```suggestion
  void goal_accepted_callback(std::shared_ptr<rclcpp_action::ServerGoalHandle<FollowJTrajAction>> goal_handle);
```

File: joint_trajectory_controller/src/joint_trajectory_controller.cpp [hunk 4]
```diff
@@ -1034,7 +1027,7 @@ rclcpp_action::CancelResponse JointTrajectoryController::cancel_callback(
   return rclcpp_action::CancelResponse::ACCEPT;
 }
 
-void JointTrajectoryController::feedback_setup_callback(
+void JointTrajectoryController::accepted_callback(
```
Comment (bmagyar) [inline joint_trajectory_controller/src/joint_trajectory_controller.cpp:1030]:
```suggestion
void JointTrajectoryController::goal_accepted_callback(
```

File: joint_trajectory_controller/src/joint_trajectory_controller.cpp [hunk 5]
```diff
@@ -776,7 +755,7 @@ controller_interface::CallbackReturn JointTrajectoryController::on_configure(
     std::string(get_node()->get_name()) + "/follow_joint_trajectory",
     std::bind(&JointTrajectoryController::goal_callback, this, _1, _2),
     std::bind(&JointTrajectoryController::cancel_callback, this, _1),
-    std::bind(&JointTrajectoryController::feedback_setup_callback, this, _1));
+    std::bind(&JointTrajectoryController::accepted_callback, this, _1));
```
Comment (bmagyar) [inline joint_trajectory_controller/src/joint_trajectory_controller.cpp:758]:
```suggestion
    std::bind(&JointTrajectoryController::goal_accepted_callback, this, _1));
```

File: joint_trajectory_controller/src/joint_trajectory_controller.cpp [hunk 6]
```diff
@@ -776,7 +755,7 @@ controller_interface::CallbackReturn JointTrajectoryController::on_configure(
     std::string(get_node()->get_name()) + "/follow_joint_trajectory",
     std::bind(&JointTrajectoryController::goal_callback, this, _1, _2),
     std::bind(&JointTrajectoryController::cancel_callback, this, _1),
```
Comment (bmagyar) [inline joint_trajectory_controller/src/joint_trajectory_controller.cpp:757]:
```suggestion
    std::bind(&JointTrajectoryController::goal_cancelled_callback, this, _1),
```

File: joint_trajectory_controller/src/joint_trajectory_controller.cpp [hunk 7]
```diff
@@ -776,7 +755,7 @@ controller_interface::CallbackReturn JointTrajectoryController::on_configure(
     std::string(get_node()->get_name()) + "/follow_joint_trajectory",
     std::bind(&JointTrajectoryController::goal_callback, this, _1, _2),
```
Comment (bmagyar) [inline joint_trajectory_controller/src/joint_trajectory_controller.cpp:756]:
```suggestion
    std::bind(&JointTrajectoryController::goal_received_callback, this, _1, _2),
```

File: joint_trajectory_controller/src/joint_trajectory_controller.cpp [hunk 8]
```diff
@@ -988,6 +966,21 @@ void JointTrajectoryController::publish_state(
...
+  }
+  // http://wiki.ros.org/joint_trajectory_controller/UnderstandingTrajectoryReplacement
+  // always replace old msg with new one for now
+  if (subscriber_is_active_)
+  {
+    add_new_trajectory_msg(msg);
+  }
+};
+
 rclcpp_action::GoalResponse JointTrajectoryController::goal_callback(
```
Comment (bmagyar) [inline joint_trajectory_controller/src/joint_trajectory_controller.cpp:984]:
```suggestion
rclcpp_action::GoalResponse JointTrajectoryController::goal_received_callback(
```

**Pull request merged**